crates/tarball/src/lib.rs

```diff
@@ -157,7 +157,7 @@ pub async fn download_tarball_to_store(
     tracing::info!(target: "pacquet::download", ?package_url, "Download completed");
 
     // TODO: benchmark and profile to see if spawning is actually necessary
-    let store_dir = Arc::new(store_dir.to_path_buf());
+    let store_dir = store_dir.to_path_buf();
```

Comment:
It's still expensive cloning. I want cheap `Arc` cloning. `store_dir` should be an `Arc<Path>`.